### PR TITLE
Fixes carousel bug

### PR DESCRIPTION
**Pull request by vasucp1207**
- Issue no https://github.com/RocketChat/RC4Community/issues/198

https://user-images.githubusercontent.com/85363195/201475252-0699b0cb-cfbc-4354-9f06-1e527c611b10.mov

### CODE REVIEW

**Review by Palanikannan1437 — CHANGES_REQUESTED** [review]
Hey @vasucp1207, it's not a good idea to commit package-lock.json, it'll cause problems for others most of the time. Please make the necessary changes to your PR

Thank you

**Comment by Palanikannan1437** [comment]
Hey @vasucp1207, sorry for not being very clear...you didn't have to delete the package-lock.json file, you just have to remove it from your PR.

One approach could be...you could revert your old commit's changes and then force push the changes again in which your `package-lock.json` isn't there in your git diff. 

Thank you 😄

**Comment by vasucp1207** [comment]
@Palanikannan1437, I do not change anything in package.json, I don't know how it changes automatically.
I commit the changes again but it appears again can you help me?

**Comment by Palanikannan1437** [comment]
> @Palanikannan1437, I do not change anything in package.json, I don't know how it changes automatically.
> I commit the changes but it appears again can you help me?

Hey yeah sure, so when you install a package using the `npm install ....` command or if it's internally run due to our existing scripts, the package.lock.json is generated

So basically while staging your files, instead of `git add .` you'd have to manually stage the files you changed and want to push.😄

**Comment by vasucp1207** [comment]
@Palanikannan1437 I remove the package-lock.json is it okay now please tell me.

**Comment by Palanikannan1437** [comment]
> @Palanikannan1437 I remove the package-lock.json is it okay now please tell me.

Sorry, but still it's in the git diff

image

> Hey @vasucp1207, sorry for not being very clear...you didn't have to delete the package-lock.json file, you just have to remove it from your PR.
> 
> One approach could be...you could revert your old commit's changes and then force push the changes again in which your `package-lock.json` isn't there in your git diff.
> 
> Thank you 😄

Please try to research and follow up on this approach of `soft resetting` and `force pushing` in git and let us know if you face any issues. Thank you!

**Comment by vasucp1207** [comment]
@Palanikannan1437 I think it's done now.

**Comment by Palanikannan1437** [comment]
Hey @vasucp1207, that was a great catch and thank you for your valuable contribution!

But currently there's no need to change the existing custom button ui/button component on our slider...instead it'd be better if you could debug and add the missing `onClick` method on our custom `PrevArrow` and `NextArrow` by going through [react-slick's custom arrow documentation](https://react-slick.neostack.com/docs/example/custom-arrows/).

Thank you and hope to see your PR get merged soon 😄

**Comment by Palanikannan1437** [comment]
> @Palanikannan1437 I think it's done now.

That was awesome and kudos on being able to remove the `package-lock.json` from your PR!! 

Hope you got to learn something out of it!!

**Comment by vasucp1207** [comment]
@Palanikannan1437, but what is the need to add the custom arrows when react-slick provides the arrows by default?

**Comment by Palanikannan1437** [comment]
> provides

Hey, I'm not sure but it was a design choice made by our contributors...and since there's just a small function they missed, there's no need to remove the entire component but instead you could add the missing function.

@Dnouv and @irffanasiff could better guide here, thank you!

**Comment by Sing-Li** [comment]
@vasucp1207   Yes.  As @Palanikannan1437   the custom arrows are BY DESIGN.   We had a designer contributor making sure that the UI elements are consistent, color schemes are attractive etc.     This is typical of all production project.    You can't use the default "just because it is there".        Please update your PR asap.

**Comment by vasucp1207** [comment]
@Sing-Li, OK

File: app/components/newscarousel.js [hunk 1]
```diff
@@ -23,6 +25,7 @@ const PrevArrow = ({ currentSlide, slideCount, ...props }) => {
 };
 
 const NextArrow = ({ currentSlide, slideCount, ...props }) => {
```
Comment (Palanikannan1437) [inline app/components/newscarousel.js:27]: You could simply de-structure the onClick method here itself! Thank you!

File: app/components/newscarousel.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import 'slick-carousel/slick/slick-theme.css';
 import Image from "next/image";
 
 const PrevArrow = ({ currentSlide, slideCount, ...props }) => {
```
Comment (Palanikannan1437) [inline app/components/newscarousel.js:7]: Likewise, you could de-structure the onClick method here itself! Thank you!

**Comment by Palanikannan1437** [comment]
LGTM @vasucp1207 !! 🥳

@Sing-Li I've reviewed the changes, please consider merging them, thank you!

**Comment by Palanikannan1437** [comment]
LGTM @vasucp1207 !! 🥳

@Sing-Li I've reviewed the changes, please consider merging them, thank you!

**Comment by Sing-Li** [comment]
Congrats @vasucp1207

**Pull request merged**